lib/samples/create_mobile_geodatabase/create_mobile_geodatabase_sample.dart

```diff
@@ -0,0 +1,353 @@
+                    children: [
+                      Text(
+                        'Number of features added: $_featureCount',
+                      ),
+                      OutlinedButton(
+                        style: OutlinedButton.styleFrom(
+                          shape: const RoundedRectangleBorder(
+                            borderRadius: BorderRadius.zero,
+                          ),
+                          side: const BorderSide(
+                            color: Colors.grey,
+                          ),
+                        ),
+                        onPressed: _displayTable,
+                        child: const Text(
+                          'View table',
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                // Display a button to create and share the mobile geodatabase.
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 10, 0, 10),
+                  child: ElevatedButton.icon(
+                    onPressed: _shareGeodatabaseUri,
+                    icon: const Icon(
+                      Icons.edit,
+                    ),
+                    label: const Text(
+                      'Create and share mobile geodatabase',
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    //_mapViewController.dispose();
+    _geodatabase?.close();
+    _geodatabase = null;
+    _featureTable = null;
+    super.dispose();
+  }
+
+  // When the map view is ready, create a map and set the viewpoint.
+  void onMapViewReady() {
+    _map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = _map;
+    _mapViewController.setViewpoint(
+      Viewpoint.withLatLongScale(
+        latitude: 41.5,
+        longitude: -100.0,
+        scale: 100000000.0,
+      ),
+    );
+    // Create the mobile geodatabase with a feature table to track
+    // location history.
+    _createGeodatabase();
+
+    setState(() => _ready = true);
+  }
+
+  // When the map is tapped, add a feature to the feature table.
+  void onTap(Offset localPosition) {
+    if (_ready == false) {
+      return;
+    }
+    final mapPoint = _mapViewController.screenToLocation(
+      screen: localPosition,
+    );
+    if (mapPoint != null) {
+      _addFeature(mapPoint);
+    }
+  }
+
+  // Create a mobile geodatabase and a feature table to store location history.
+  void _createGeodatabase() async {
+    final directory = await getApplicationDocumentsDirectory();
+    final geodatabaseFile = File(
+        '${directory.path}${Platform.pathSeparator}localHistory.geodatabase');
+    if (geodatabaseFile.existsSync()) {
+      geodatabaseFile.deleteSync();
+    }
+    _geodatabase?.close();
+
+    Geodatabase.create(fileUri: geodatabaseFile.uri).then(
```

Comment:
We should use `await` as necessary to make sure the geodatabase and its table are created before `_ready` is set to `true`. It would be simpler to do that with a try/catch rather than `.then()` and `.onError()`. Something like:

```dart
try {
  _geodatabase = await Geodatabase.create(...);
  await _createGeodatabaseFeatureTable();
} catch (e) {
  _showDialog(...);
}
```

And then similarly in `_createGeodatabaseFeatureTable()`, it could have:

```dart
// ...
_featureTable = await _geodatabase!.createTable(...);
// ...
```

`_createGeodatabaseFeatureTable()` wouldn't need a second try/catch block, since it would be covered by the try/catch block in `_createGeodatabase()`.